nixos/modules/services/web-apps/bookstack.nix

```diff
@@ -69,139 +230,129 @@ in {
-        default = null;
-        example = "/run/keys/bookstack-mailpassword";
-        description = ''
-          A file containing the password corresponding to
-          {option}`mail.user`.
-        '';
-      };
-      encryption = mkOption {
-        type = with types; nullOr (enum [ "tls" ]);
-        default = null;
-        description = "SMTP encryption mechanism to use.";
+    settings = lib.mkOption {
+      default = { };
+      description = ''
+        Options for Bookstack configuration. Refer to
+        <https://github.com/BookStackApp/BookStack/blob/development/.env.example> for
+        details on supported values. For passing secrets, append "_FILE" to the
+        setting name. For example, you may create a file `/var/secrets/db_pass.txt`
+        and set `services.bookstack.settings.DB_PASSWORD_FILE` to `/var/secrets/db_pass.txt`
+        instead of providing a plaintext password using `services.bookstack.settings.DB_PASSWORD`.
+      '';
+      example = lib.literalExpression ''
+        {
+          APP_ENV = "production";
+          APP_KEY_FILE = "/var/secrets/bookstack-app-key.txt";
+          DB_HOST = "db";
+          DB_PORT = 3306;
+          DB_DATABASE = "bookstack";
+          DB_USERNAME = "bookstack";
+          DB_PASSWORD_FILE = "/var/secrets/bookstack-mysql-password.txt";
+        }
+      '';
+      type = lib.types.submodule {
+        freeformType = lib.types.attrsOf (
+          lib.types.oneOf [
+            lib.types.str
+            lib.types.int
+            lib.types.bool
+          ]
+        );
+        options = {
+          DB_PORT = lib.mkOption {
+            type = lib.types.int;
+            default = 3306;
+            description = ''
+              The port your database is listening at.
+            '';
+          };
+          DB_HOST = lib.mkOption {
+            type = lib.types.str;
+            default = "localhost";
+            description = ''
+              The machine which hosts your database.
+            '';
+          };
+          DB_PASSWORD_FILE = lib.mkOption {
+            type = lib.types.path;
+            description = ''
+              The file containing your mysql/mariadb database password.
+            '';
+            example = "/var/secrets/bookstack-mysql-pass.txt";
+          };
+          APP_KEY_FILE = lib.mkOption {
+            type = lib.types.path;
+            description = ''
+              The path to your appkey. The file should contain a 32 character
+              random app key. This may be set using `echo "base64:$(head -c 32
+              /dev/urandom | base64)" > /path/to/key-file`.
+            '';
+          };
+          APP_URL = lib.mkOption {
+            type = lib.types.str;
+            default =
+              if cfg.hostname == "localhost" then "http://${cfg.hostname}" else "https://${cfg.hostname}";
+            defaultText = ''
+              http(s)://''${config.services.bookstack.hostname}
+            '';
```

Comment:
```suggestion
            defaultText = "http(s)://''${config.services.bookstack.hostname}";
```

Reply:
This breaks building the manual.

Reply:
It shouldn't. We just remove some extra new lines which look weird in the manual.